<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>PrediXcan</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
</head>

<body class="stackedit">
  <div class="stackedit__html"><h2 id="predixcan">PrediXcan</h2>
<h3 id="to-get-the-snp-annotation-file">To get the SNP annotation file</h3>
<p>To provide SNPID, pos, ref, and alt allele information</p>
<ul>
<li>add rsid from dbsnp150<br>
Note:<br>
<em>37114456 SNPs in dbsnp150</em><br>
*425583 out of 7034069 SNPs with no rsIDs/bialleic SNPs, and these SNPs will be ignored during the training</li>
</ul>
<p><a href="https://s3.amazonaws.com/predictdb2/GTEx-V7_HapMap-2017-11-29_README.txt">https://s3.amazonaws.com/predictdb2/GTEx-V7_HapMap-2017-11-29_README.txt</a><br>
<em>SNPs were then filtered to those<br>
with minor allele frequency &gt;= 1%, R2 &gt;= 0.8, biallelic only, and those which<br>
would be unambiguously stranded, i.e. the polymorphism is not to the base’s<br>
complement.</em></p>
<pre class=" language-bash"><code class="prism  language-bash">python get_rsid.py \
/projects/b1047/zhong/software/utility/rsid_SNP_b150.txt \
/projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/input/annotations/snp_annotation/annotation3.txt
/projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/input/annotations/snp_annotation/ \
annotation_snp


<span class="token punctuation">[</span>yzu280@quser11 snp_annotation<span class="token punctuation">]</span>$ <span class="token function">awk</span> <span class="token string">'<span class="token variable">$6</span>=="NA" {print}'</span> annotation_snp.txt <span class="token operator">|</span> <span class="token function">wc</span> -l
425583

</code></pre>
<ul>
<li>make sure the reference and alternative allele matches with the dosage files.<br>
[yzu280@quser13 scripts]$ python check_allele.py</li>
</ul>
<h3 id="split-the-snp-files-by-chr">split the SNP files by chr</h3>
<pre class=" language-python"><code class="prism  language-python"><span class="token comment">#Skip non-single letter polymorphisms</span>
 <span class="token number">53</span>             <span class="token keyword">if</span> <span class="token builtin">len</span><span class="token punctuation">(</span>refAllele<span class="token punctuation">)</span> <span class="token operator">&gt;</span> <span class="token number">1</span> <span class="token operator">or</span> <span class="token builtin">len</span><span class="token punctuation">(</span>effectAllele<span class="token punctuation">)</span> <span class="token operator">&gt;</span> <span class="token number">1</span><span class="token punctuation">:</span>
 <span class="token number">54</span>                 <span class="token keyword">continue</span>
 <span class="token number">55</span>             <span class="token comment"># Skip ambiguous strands</span>
 <span class="token number">56</span>             <span class="token keyword">if</span> SNP_COMPLEMENT<span class="token punctuation">[</span>refAllele<span class="token punctuation">]</span> <span class="token operator">==</span> effectAllele<span class="token punctuation">:</span>
 <span class="token number">57</span>                 <span class="token keyword">continue</span>
 <span class="token number">58</span>             <span class="token keyword">if</span> rsid <span class="token operator">==</span> <span class="token string">'NA'</span><span class="token punctuation">:</span>
 <span class="token number">59</span>                 <span class="token keyword">continue</span>
</code></pre>
<pre class=" language-bash"><code class="prism  language-bash"><span class="token punctuation">[</span>yzu280@quser11 scripts<span class="token punctuation">]</span>$ python split_snp_annot_by_chr.py  \
/projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/input/annotations/snp_annotation/annotation_snp.txt  \
/projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/input/annotations/snp_annotation/annotation_snp
</code></pre>
<p>Before filtering, the number of SNPs is 7034069<br>
After filtering, the number of SNPs is 5189842</p>
<h3 id="split-the-genotype-by-chr">split the genotype by chr</h3>
<p>Note: need to use the annotation files after filtering</p>
<pre class=" language-bash"><code class="prism  language-bash">python split_genotype_by_chr.py /projects/b1047/zhong/Hepatocyte_project/data/genotype/eQTL/condition1_expression_tmm_normalization_dosage_v7_2.txt \ /projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/input/genotypes/genotype_filter \
/projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/input/annotations/snp_annotation/annotation_snp.filterd.txt
</code></pre>
<ul>
<li class="task-list-item"><input type="checkbox" class="task-list-item-checkbox" disabled=""> the genotype number is 5189842</li>
</ul>
<h3 id="get-the-gene-expression-annotationscovariate-file">get the gene expression annotations/covariate file</h3>
<ul>
<li>Transpose the expression and covariates</li>
<li>Covariates include: 10 PEER, sex, platform, batch, PC1-3</li>
</ul>
<h3 id="start-the-training">Start the training</h3>
<h4 id="test-the-predixcan-model-built-with-white-people">Test the PrediXcan model built with white people</h4>
<pre class=" language-bash"><code class="prism  language-bash">python convert_plink_to_dosage.py \
-b /projects/b1047/zhong/Hepatocyte_project/data/genotype/baseline.genotype.eqtl.excludeoutlier.header \
-o /projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/aa_input/aa_genotype -p plink


python PrediXcan.py --predict \
--dosages /projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/aa_input/ \
--dosages_prefix aa_genotype \
--sample /projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/aa_input/aa_genotype.nosex  \
--weights <span class="token punctuation">..</span>/Liver/gtex_v7_Liver_imputed_europeans_tw_0.5_signif.db \
--output_prefix /projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/aa_input/

</code></pre>
<pre class=" language-r"><code class="prism  language-r"><span class="token operator">&gt;</span> quantile<span class="token punctuation">(</span>cor_exp<span class="token punctuation">,</span> c<span class="token punctuation">(</span>.<span class="token number">0</span><span class="token punctuation">,</span>.<span class="token number">25</span><span class="token punctuation">,</span>.<span class="token number">5</span><span class="token punctuation">,</span>.<span class="token number">75</span><span class="token punctuation">,</span><span class="token number">1</span><span class="token punctuation">)</span><span class="token punctuation">,</span> na.rm <span class="token operator">=</span> T<span class="token punctuation">)</span>
        <span class="token number">0</span>%        <span class="token number">25</span>%        <span class="token number">50</span>%        <span class="token number">75</span>%       <span class="token number">100</span>% 
<span class="token operator">-</span><span class="token number">0.3752232</span> <span class="token operator">-</span><span class="token number">0.0263705</span>  <span class="token number">0.0792258</span>  <span class="token number">0.1993307</span>  <span class="token number">0.9068370</span> 
<span class="token comment"># &gt; length(exp_id)</span>
<span class="token comment"># [1] 23042</span>
<span class="token comment"># &gt; length(predicted_id)</span>
<span class="token comment"># [1] 3351</span>
</code></pre>
<p>Note: The correlation is low.<br>
possible explanations are:</p>
<ol>
<li>less SNPs are used for GTEx model</li>
<li>Should compare the expression after regressing out the covariates.</li>
<li></li>
</ol>
<p>after regressing out the covariates, the correlation is actually less</p>
<pre class=" language-r"><code class="prism  language-r"><span class="token operator">&gt;</span> quantile<span class="token punctuation">(</span>cor_exp<span class="token punctuation">,</span> c<span class="token punctuation">(</span>.<span class="token number">0</span><span class="token punctuation">,</span>.<span class="token number">25</span><span class="token punctuation">,</span>.<span class="token number">5</span><span class="token punctuation">,</span>.<span class="token number">75</span><span class="token punctuation">,</span><span class="token number">1</span><span class="token punctuation">)</span><span class="token punctuation">,</span> na.rm <span class="token operator">=</span> T<span class="token punctuation">)</span>
         <span class="token number">0</span>%         <span class="token number">25</span>%         <span class="token number">50</span>%         <span class="token number">75</span>%        <span class="token number">100</span>% 
<span class="token operator">-</span><span class="token number">0.38839424</span> <span class="token operator">-</span><span class="token number">0.02202818</span>  <span class="token number">0.08649508</span>  <span class="token number">0.22052888</span>  <span class="token number">0.76986455</span> 
<span class="token comment">#</span>
</code></pre>
<h3 id="test-the-predixcan-model-built-on-white-people-on-gtex-liver-aa-people">Test the PrediXcan model built on white people on GTEx Liver AA people</h3>
<pre class=" language-bash"><code class="prism  language-bash">python convert_plink_to_dosage.py \
-b /projects/b1047/zhong/v7_gtex/genotype/GTEx_Analysis_2016-01-15_v7_WholeGenomeSeq_635Ind_PASS_AB02_GQ20_HETX_MISS15_PLINKQC_maf0.05_liver \
-o /projects/b1047/zhong/Hepatocyte_project/Yilin/PrediXcan/data/aa_input/gtex_liver \
-p plink
</code></pre>
</div>
</body>

</html>
